extractor/src/main/java/org/schabi/newpipe/extractor/services/soundcloud/extractors/SoundcloudCommentsInfoItemExtractor.java

```diff
@@ -77,6 +143,54 @@ public String getUrl() {
 
     @Override
     public String getThumbnailUrl() {
-        return json.getObject("user").getString("avatar_url");
+        return user.getString("avatar_url");
+    }
+
+    @Override
+    public Page getReplies() {
+        if (replyCount == CommentsInfoItem.UNKNOWN_REPLY_COUNT) {
+            final JsonArray replies = new JsonArray();
+            final CommentsInfoItemsCollector collector = new CommentsInfoItemsCollector(
+                    ServiceList.SoundCloud.getServiceId());
+            // SoundCloud has only comments and top level replies, but not nested replies.
+            // Therefore, replies cannot have further replies.
+            if (superComment == null) {
+                // Loop through all comments which come after the original comment
+                // to find its replies.
+                final JsonArray allItems = json.getArray(SoundcloudCommentsExtractor.COLLECTION);
+                boolean foundReply = false;
+                for (int i = index + 1; i < allItems.size(); i++) {
+                    final JsonObject comment = allItems.getObject(i);
+                    if (SoundcloudParsingHelper.isReplyTo(item, comment)) {
+                        replies.add(comment);
+                        collector.commit(new SoundcloudCommentsInfoItemExtractor(
+                                json, i, comment, url, item));
+                        foundReply = true;
+                    } else if (foundReply) {
+                        // Only the comments directly after the original comment
+                        // having the same timestamp are replies to the original comment.
+                        // The first comment not having the same timestamp
+                        // is the next top-level comment.
+                        break;
+                    }
+                }
+            }
+            replyCount = replies.size();
+            if (collector.getItems().isEmpty()) {
```

Comment:
I guess you are collecting items here to make sure not to show that there are replies when actually all replies give an error when collected.
- Since you are already collecting replies here, can't you set the items as the `repliesPage` content directly instead of passing the `json` again? Or maybe make a `Serializable` structure of some sort. To avoid recollecting them again later.
- Or even better, I would skip this check completely and avoid collecting items beforehand. If one of the comments yields an error while being collected, then the error would be shown in the frontend and it is expected that in case of error on an item the list of replies might be empty.